product_pricelist_alternative/models/product_pricelist.py

```diff
@@ -74,3 +75,18 @@ def _compute_price_rule(self, products, qty, uom=None, date=False, **kwargs):
                     if alternative_price_rule[product.id][0] < res[product.id][0]:
                         res[product.id] = alternative_price_rule[product.id]
         return res
+
+    @api.constrains("alternative_pricelist_ids")
+    def _check_pricelist_alternative_items_based_on_other_pricelist(self):
+        """Alternative pricelists can not contain items based on other pricelist"""
+        for pricelist in self:
+            if pricelist.alternative_pricelist_ids.item_ids.filtered(
+                lambda item: item.compute_price == "formula"
+                and item.base == "pricelist"
+            ):
```

Comment:
Non-blocking: you can use `any()` instead of `filtered` for better performances:
```suggestion
            if any(
                i.compute_price == "formula" and i.base == "pricelist"
                for i in pricelist.alternative_pricelist_ids.item_ids
            ):
```
The reason is:
- `filtered()` loops over each and every the record in the recordset, and only after going through them all the `if` condition is evaluated on the resulting recordset
- `any()` stops as soon as the iterator passed as argument returns a `True`-ish value